native/src/main/java/io/ballerina/persist/objects/Relation.java

```diff
@@ -0,0 +1,113 @@
+/*
+ *  Copyright (c) 2022, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ *  WSO2 Inc. licenses this file to you under the Apache License,
+ *  Version 2.0 (the "License"); you may not use this file except
+ *  in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing,
+ *  software distributed under the License is distributed on an
+ *  "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ *  KIND, either express or implied.  See the License for the
+ *  specific language governing permissions and limitations
+ *  under the License.
+ */
+package io.ballerina.persist.objects;
+
+import java.util.ArrayList;
+import java.util.Optional;
+
+/**
+ * Class to store entity relations.
+ *
+ * @since 0.1.0
+ */
+
+public class Relation {
+
+    private final ArrayList<String> keyColumns;
+    private final ArrayList<String> references;
+
+    private ArrayList<FieldMetaData> relatedFields = new ArrayList<>();
+    private final boolean isChild;
```

Comment:
Instead of having this `isChild` variable, shall we have 2 `Relation` arrays in the Entity for child relations and parent relations?
```
private ArrayList<Relation> childRelations = new ArrayList<>();
private ArrayList<Relation> parentRelations = new ArrayList<>();
```

Reply:
I checked the code for implementing this, for most of the parts of the code (except update and create methods mainly), code depends on the inclusion of the related entity fields, rather than their position of the relation. if we include them in two Arraylists, it might unnecessarily lengthen the code with repeating code segments for two arraylists.